daq/acl_state_collector.py

```diff
@@ -0,0 +1,70 @@
+"""Process ACL states"""
+
+from __future__ import absolute_import
+
+import logger
+
+LOGGER = logger.get_logger('acl')
+
+
+class AclStateCollector:
+    """Processing ACL states for ACL counting"""
+
+    def __init__(self):
+        self._switch_configs = {}
+
+    def get_port_acl_count(self, switch, port, port_acl_samples):
+        """Return the ACL count for a port"""
+
+        switch_config = self._switch_configs.get(switch)
+        if not switch_config:
+            LOGGER.warning('Switch not defined in Faucet dps config: %s', switch)
+            return {}
```

Comment:
rather than return empty, I think there should be a mechanism that includes an error indicator in the result. All of the error cases here seem fairly fundamental and shouldn't happen in normal use, correct?

Reply:
Updated. Also added protobuf message to define the output of the function. The `expected_acl_count` in test_acl_state_collector gives an idea about what the output looks like. PTAL..